<?xml version="1.0"?>
<launch>

  <node name="transform_point_cloud" pkg="transform_point_cloud"
      type="transform_point_cloud"
      output="screen" >
    <param name="target_frame" value="frame1" />
  </node>

  <node name="transform_point_cloud_py" pkg="transform_point_cloud"
      type="transform_point_cloud_py.py"
      output="screen" >
    <param name="target_frame" value="frame2" />
    <remap from="point_cloud" to="transform_point_cloud/point_cloud" />
  </node>

  <node name="generate_point_cloud" pkg="transform_point_cloud"
      type="generate_point_cloud.py"
      output="screen" ns="transform_point_cloud" >
  </node>

  <node pkg="tf2_ros" type="static_transform_publisher" name="static_frame1"
      args="1 2 2 0.1 0.2 0.3 map frame1" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="static_frame2"
      args="0 0 -3.0 0.0 0.0 0.25 frame1 frame2" />

  <node name="rviz" pkg="rviz" type="rviz"
      args="-d $(find transform_point_cloud)/config/pc2.rviz"
      required="true"
      output="screen">
  </node>

  <node name="rqt" pkg="rqt_gui" type="rqt_gui"
      args="--perspective-file $(find transform_point_cloud)/config/lookup.perspective"
      required="true"
      output="screen">
  </node>

</launch>
